shared/middleware/helpers/table.rb

```diff
@@ -19,7 +20,7 @@ def items()
   end
   
   def delete(id)
-    delete_count = items.where(row_id:id).delete
+    delete_count = @can_modify ? items.where(row_id:id).delete : 0
```

Comment:
This is an optional, style/design thing, but I tend to write my code using a pattern of throwing out the edge cases first with the goal, generally, to have the final action be the result I want to return. I do this because I think for readers of the code (and correct me if you disagree) it helps them sort of check off their list of edge cases as they understand the code.

So I'd probably have done:

```
def delete(id)
  return 0 unless @can_modify
  ... leave the rest as-before
end
```

This, to me, tells me that after the first line, that I don't have to think about read-only users anymore, at all. So I move to the next line with less cognitive load.

(I said this was optional!)